### PR TITLE
Fixed bug where BlockBlobClient and PageBlobClient would throw NullRe…

### LINKED ISSUE #13232: [BUG] new BlockBlobClient(uri) will fail with 'Object reference not set to an instance of an object.'
**Describe the bug**
In the just released Azure.Storage.Blobs 12.5.0-preview.5：
Run `new BlockBlobClient(uri)` will fail with 'Object reference not set to an instance of an object.'
When you input BlobClientOption as not null, the error won't happen.

**Expected behavior**
What is the expected behavior?
No Error, BlockblobClient should be created

**Actual behavior (include Exception or Stack Trace)**
What is the actual behavior?
Throw exception:

> System.NullReferenceException
>   HResult=0x80004003
>   Message=Object reference not set to an instance of an object.
>   Source=Azure.Storage.Blobs
>   StackTrace:
>    at Azure.Storage.Blobs.Specialized.BlockBlobClient.AssertNoClientSideEncryption(BlobClientOptions options)
>    at testTracking.Program.Main(String[] args) in C:\Users\weiwei\source\repos\testTracking\testTracking\Program.cs:line 36

**To Reproduce**
Steps to reproduce the behavior (include a code snippet, screenshot, or any additional information that might help us reproduce the issue)

1. Get a blob Uri like "https://***.blob.core.windows.net/test/blob1"
2. run `new BlockBlobClient(uri)`

**Environment:**
 - Name and version of the Library package used: [e.g. Azure.Storage.Blobs 12.5.0-preview.5] 
 - Hosting platform or OS and .NET runtime version (`dotnet --info` output for .NET Core projects): [e.g. Azure AppService or Windows 10 .NET Framework 4.8]
```
.NET Core SDK (reflecting any global.json):
 Version:   2.2.108
 Commit:    33ed5b90ce

Runtime Environment:
 OS Name:     Windows
 OS Version:  10.0.18363
 OS Platform: Windows
 RID:         win10-x64
 Base Path:   C:\Program Files\dotnet\sdk\2.2.108\

Host (useful for support):
  Version: 2.2.6
  Commit:  7dac9b1b51

.NET Core SDKs installed:
  2.2.108 [C:\Program Files\dotnet\sdk]

.NET Core runtimes installed:
  Microsoft.AspNetCore.All 2.2.6 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.All]
  Microsoft.AspNetCore.App 2.2.6 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 2.2.6 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]

To install additional .NET Core runtimes or SDKs:
  https://aka.ms/dotnet-download
```
 - IDE and version : [e.g. Visual Studio 16.3]
VS 15.7.4

### PR DESCRIPTION
**Pull request by seanmcc-msft**
…ferenceExceptions when using Uri constructor

Resolves https://github.com/Azure/azure-sdk-for-net/issues/13232

### CODE REVIEW

File: sdk/storage/Azure.Storage.Blobs/tests/AppendBlobClientTests.cs [hunk 1]
```diff
@@ -49,6 +49,27 @@ public void Ctor_ConnectionString()
             Assert.AreEqual("accountName", builder.AccountName);
         }
 
+        [Test]
```
Comment (blueww) [inline sdk/storage/Azure.Storage.Blobs/tests/AppendBlobClientTests.cs:52]: Please make sure all kinds of blob client can be created with all kinds credential, and have test cases to cover them.

Comment (seanmcc-msft) [reply sdk/storage/Azure.Storage.Blobs/tests/AppendBlobClientTests.cs:52]: I did.

Comment (seanmcc-msft) [reply sdk/storage/Azure.Storage.Blobs/tests/AppendBlobClientTests.cs:52]: There are test cases for BlobClient, BlobBaseClient, AppendBlobClient, and PageBlobClient.

Comment (blueww) [reply sdk/storage/Azure.Storage.Blobs/tests/AppendBlobClientTests.cs:52]: Please confirm  sure all kinds  of credential are covered, and BlockBlobClient also covered.

Comment (seanmcc-msft) [reply sdk/storage/Azure.Storage.Blobs/tests/AppendBlobClientTests.cs:52]: https://github.com/Azure/azure-sdk-for-net/pull/13332/files#diff-85f647442c3573f2903357a126170dacR60

Comment (blueww) [reply sdk/storage/Azure.Storage.Blobs/tests/AppendBlobClientTests.cs:52]: Hi Sean, 
The above link is for BlockblobClient.
Please confirm all kinds of credentail are covered by all kinds of blob clients.
(I see the test for Uri only, please make sure oauth, key, connection string are all covered)

**Review by kasobol-msft — APPROVED** [review]

**Pull request merged**